src/app/checkout/checkout.tpl.html

```diff
@@ -82,8 +82,8 @@ <h3 class="panel-name inline-block" translate>Secure Checkout</h3>
         <checkout-cart-summary cart-data="$ctrl.cartData"
                                show-submit-btn="$ctrl.checkoutStep === 'review'"
                                enable-submit-btn="$ctrl.enableSubmitOrderBtn"
-                               on-submit="$ctrl.submittedOrder = true"
-                               submitting-order="$ctrl.submittingOrder">
+                               submitting-order="$ctrl.submittingOrder"
+                               component-reference="$ctrl.selfReference">
```

Comment:
I get why you are passng down the reference, but I don't understand how we move the user forward through to the next stage in the checkout.

Reply:
The `cart-summary` component is emitting an event on the `$rootScope` of `componentReference` and the `step-3` component is listening for that event and calling the submit logic.